### DISCUSSION TITLE
Android to Android - Long press not working

### DISCUSSION THREAD

**Discussion by vrepub** [rustdesk/rustdesk#2594] [Q&A]
I cannot get long press to work.  Is there a setting I have to adjust or something?

No long press controlling android 11 from android 9, and no long press controlling android 9 from android 11.

Thanks

**Comment by rustdesk** [comment]
@Heap-Hop

**Comment by Heap-Hop** [comment]
Hi, @vrepub , thanks for your report. 
It is a good discussion about sending long press to remote Android, but unfortunately we haven't found a perfect solution to do that.
Now we use Android AccessibilityService to receive the input commands, but AccessibilityService could only receive a **complete sequence** of input. 
That means it only work with like:
> tap down -> hold(long press several secs or mins) -> tap up

Android AccessibilityService only process when you **tap up**.

You have to do ALL long press actions(including up your finger) beforehand.
It will make long press **difficult and confusing**.

Currently, we implement a behavior that automatically send a `tap up` action when you press and hold seconds. It is sometimes useful in some scenarios where a long press triggers a menu. But lost the ability to `long press and move`.

We may add a option to switch `automatically tap up` and `complete sequence but confusing long press` modes. 
Both are imperfect :(

We would like to collect more suggestions, or new solutions.

**Comment by jpy920718** [comment]
这是来自QQ邮箱的假期自动回复邮件。 您好，我最近正在休假中，无法亲自回复您的邮件。我将在假期结束后，尽快给您回复。
